pkg/build/admission/jenkinsbootstrapper/admission.go

```diff
@@ -0,0 +1,159 @@
+package jenkinsbootstrapper
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	coreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/unversioned"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	kutilerrors "k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/openshift/origin/pkg/api/latest"
+	authenticationclient "github.com/openshift/origin/pkg/auth/client"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	jenkinscontroller "github.com/openshift/origin/pkg/build/controller/jenkins"
+	"github.com/openshift/origin/pkg/client"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/config/cmd"
+)
+
+func init() {
+	admission.RegisterPlugin("JenkinsBootstrapper", func(c clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewJenkingsBootstrapper(c.Core()), nil
+	})
+}
+
+type jenkingsBootstrapper struct {
+	*admission.Handler
+
+	privilegedRESTClientConfig restclient.Config
+	serviceClient              coreclient.ServicesGetter
+	openshiftClient            client.Interface
+
+	jenkinsConfig configapi.JenkinsPipelineConfig
+}
+
+// NewJenkingsBootstrapper returns an admission control for builds that overrides
+// settings on builds
+func NewJenkingsBootstrapper(serviceClient coreclient.ServicesGetter) admission.Interface {
+	return &jenkingsBootstrapper{
+		Handler:       admission.NewHandler(admission.Create),
+		serviceClient: serviceClient,
+	}
+}
+
+// Admit appplies configured overrides to a build in a build pod
+func (a *jenkingsBootstrapper) Admit(attributes admission.Attributes) error {
+	if a.jenkinsConfig.Enabled != nil && !*a.jenkinsConfig.Enabled {
+		return nil
+	}
+	if len(attributes.GetSubresource()) != 0 {
+		return nil
+	}
+	if attributes.GetResource().GroupResource() != buildapi.Resource("buildconfigs") && attributes.GetResource().GroupResource() != buildapi.Resource("builds") {
+		return nil
+	}
+	if !needsJenkinsTemplate(attributes.GetObject()) {
+		return nil
```

Comment:
> maybe glog to help debugging this three if's if somethiing terrible happens?

Seems like that would be pretty noisy.  This if block is a common case where you're making a non-jenkins build and it wouldn't help someone trying to use the API since he wouldn't be able to see the message.

If you still want it, let me know and I'll at it at level 4.